client/landing/stepper/declarative-flow/new-hosted-site-flow.ts

```diff
@@ -136,9 +138,12 @@ const hosting: Flow = {
 			() => {
 				if ( currentStepSlug === undefined ) {
 					resetOnboardStore();
+					if ( query.get( 'campaign' ) === 'reddit' ) {
```

Comment:
@donlair w.r.t using a query param to force the free trial to be available (pdKhl6-2SQ-p2#comment-4897) I was thinking of a query parameter like `?campaign=reddit`. Does that seem reasonable? We can change it to whatever, maybe you'd like to pass some sort of campaign ID so you can try different campaigns?

Also, will the campaigns go through the LOHP or straight to the onboarding flow? That is to say, will we send traffic to `wordpress.com/hosting?campaign=reddit` first or will they go straight to `wordpress.com/setup/new-hosted-site?campaign=reddit`?

Reply:
Good questions, thanks Philip!

> w.r.t using a query param to force the free trial to be available (pdKhl6-2SQ-p2#comment-4897) I was thinking of a query parameter like ?campaign=reddit. Does that seem reasonable? We can change it to whatever, maybe you'd like to pass some sort of campaign ID so you can try different campaigns?

Hmm, it would be nice to be able to pass additional info if we could use that info as props in the Tracks events. Something like: `?source=reddit&campaignid=[string]`. Or it could be condensed into something like `?campaign=reddit_[campaignNameString]`. Not necessary, but that would help with campaign analysis post-launch.

I'm fine with `?campaign=reddit` as a v1 and adding more tracking-related features later if/when we start scaling the campaign.

> Also, will the campaigns go through the LOHP or straight to the onboarding flow? That is to say, will we send traffic to wordpress.com/hosting?campaign=reddit first or will they go straight to wordpress.com/setup/new-hosted-site?campaign=reddit?

The user will land on a marketing page first and a CTA on that page will point the user to the hosting flow. That landing page may be a traditional marketing landing page built in Landpack or an article-style landing page posted on /go. 

Either way we'll have control over the url in the CTA and can add the query params to the link.

Reply:
> I'm fine with ?campaign=reddit as a v1 and adding more tracking-related features later if/when we start scaling the campaign.

Sounds good 👍

> we'll have control over the url in the CTA

Perfect thanks!